src/components/UpdateInput.tsx

```diff
@@ -1,14 +1,17 @@
+import { useRouter } from "next/router";
```

Comment:
You should remove the business logic from this component, and use the `business` and `display` folder.

So basically:

#### `/business/UpdateIdentity`
Which passes the async function used by the form component on submit as props to its children.

#### `/display/UpdateIdentityForm`
Which only display the css.

Reply:
We could go even further and wrap `UpdateIdentityForm` with the add and remove business components. But it would require a bit of logic in the display component then (like the `preventDefault`).

I think the first one is better. @Fenntasy what do you think?

Reply:
I just had the discussion with fenn, and we will use my second proposition, with a twist.

First, regarding the `preventDefault`, it is in fact the form responsibility to control its re-render. In other word, the business logic component do not care what kind of children they have, they only care about passing their logic. So the `preventDefault` would stay in the display component.

Like we said previously, the update identity flow is just a combination of `add` and `remove`. We already have those business component.

The update identity business component role would be to keep the current identity until it gets deleted.

So basically, `<UpdateIdentity/>` would be scoped inside the 3 business components.

Reply:
OK done! 0733f6a a8f1a4a e2787e7